planner/core/rule_join_elimination.go

```diff
@@ -29,35 +28,52 @@ type outerJoinEliminator struct {
 // 2. outer join elimination with duplicate agnostic aggregate functions: For example left outer join.
 //    If the parent only use the columns from left table with 'distinct' label. The left outer join can
 //    be eliminated.
-func (o *outerJoinEliminator) tryToEliminateOuterJoin(p *LogicalJoin, aggCols []*expression.Column, parentSchema *expression.Schema) LogicalPlan {
+func (o *outerJoinEliminator) tryToEliminateOuterJoin(p *LogicalJoin, aggCols []*expression.Column, parentSchema *expression.Schema) (LogicalPlan, error) {
 	var innerChildIdx int
 	switch p.JoinType {
 	case LeftOuterJoin:
 		innerChildIdx = 1
 	case RightOuterJoin:
 		innerChildIdx = 0
 	default:
-		return p
+		return p, nil
 	}
 
 	outerPlan := p.children[1^innerChildIdx]
 	innerPlan := p.children[innerChildIdx]
 	// outer join elimination with duplicate agnostic aggregate functions
-	if o.isAggColsAllFromOuterTable(outerPlan, aggCols) {
-		return outerPlan
+	matched, err := o.isAggColsAllFromOuterTable(outerPlan, aggCols)
+	if err != nil {
+		return nil, err
+	}
+	if matched {
+		return outerPlan, nil
 	}
 	// outer join elimination without duplicate agnostic aggregate functions
-	if !o.isParentColsAllFromOuterTable(outerPlan, parentSchema) {
-		return p
+	matched, err = o.isParentColsAllFromOuterTable(outerPlan, parentSchema)
+	if err != nil {
```

Comment:
It's better to return `p, err` in the judgment.
Then we can merge the two `if` by:
```go
if err != nil || !matched {
	return p, err
}
```

Reply:
Yes, it's better.